torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -429,8 +430,12 @@ class FullyShardedDataParallel(nn.Module):
     .. warning::
         Module should be already placed on the destination device or
         device is set properly using ``torch.cuda.set_device(device_id)``.
+        Alternatively, if module is input on CPU, ``device_id`` argument can
+        be specified to move module to CUDA device during initialization.
         FSDP will get compute device from module first, if module device
```

Comment:
nit: Since these warnings are user-visible in the docs, I am more inclined to be picky about style/grammar. Since I am the one who is picky, I can make a pass over the user-visible documentation before branch cut.

E.g.
- Maybe we should backtick `module` (i.e. \`\`module\`\`) since it specifically refers to the `module` argument in the constructor.
- Some grammatical changes:

> If the destination CUDA device has ID `dev_id`, either (1) `module` should already be placed on that device, (2) the device should be set using `torch.cuda.set_device(dev_id)`, or (3) `dev_id` should be passed into the `device_id` constructor argument. This FSDP instance's compute device is CPU only if `module` is input on CPU and the `device_id` argument is not specified. Otherwise, the compute device is the CUDA device with ID `dev_id`.

Reply:
sure, we can work together to make sure the user facing documentation is as clear as possible.